posthog/auth.py

```diff
@@ -94,7 +94,12 @@ def authenticate(cls, request: Union[HttpRequest, Request]) -> Optional[Tuple[An
         assert personal_api_key_object.user is not None
 
         # :KLUDGE: CHMiddleware does not receive the correct user when authenticating by api key.
-        tag_queries(user_id=personal_api_key_object.user.pk, team_id=personal_api_key_object.user.current_team_id)
+        tag_queries(
+            user_id=personal_api_key_object.user.pk,
+            team_id=personal_api_key_object.user.current_team_id,
+            access_method="personal_api_key",
+        )
+        request.using_personal_api_key = True  # type: ignore
```

Comment:
I wonder if we should get these lines in first in a separate PR, and monitor for a bit. We spotted 5 users on `/api/event` yesterday, out of whom only the one troubled user would be affecter. However are there perhaps a few dozen users using `/api/project/x/event` that we don't know of, who'd also be affected?

Reply:
https://github.com/PostHog/posthog/pull/13443